### PR TITLE
AndroidX libraries and react-native@0.60.0+ support

### PR DESCRIPTION
**Pull request by oleksandr-dziuban**
- AndroidX libraries support added
- Fixed Android release build issues with old File Provider libraries
- react-native@0.60.0+ support and auto linking support added
- Updated ReadMe file with usage for eact-native@0.60.0+

### CODE REVIEW

**Comment by jgcmarins** [comment]
Awesome!
We're facing flow errors `Check for errors in the example app using Flow`. Could you take a look please?
I would also ask you to update [example](https://github.com/react-native-community/react-native-share/tree/master/example)

**Comment by oleksandr-dziuban** [comment]
> example

Sure, working on it now, forgot to update example

**Comment by oleksandr-dziuban** [comment]
> Awesome!
> We're facing flow errors `Check for errors in the example app using Flow`. Could you take a look please?
> 
> Flow Errors (click to expand)
> I would also ask you to update [example](https://github.com/react-native-community/react-native-share/tree/master/example)

@jgcmarins Fixed, please check

**Comment by MateusAndrade** [comment]
AWESOME! :tada:  :tada: 

Thanks for helping us @oleksandr-dziuban. I gave a quick look, and It LGTM.

**Comment by oleksandr-dziuban** [comment]
> AWESOME! 🎉 🎉
> 
> Thanks for helping us @oleksandr-dziuban. I gave a quick look, and It LGTM.
> 
> Im going to give a quick test at this PR as soon as possible, so we can Merge this. 😄

Thanks, waiting publishing version 2.0.0 👍

**Comment by oleksandr-dziuban** [comment]
@MateusAndrade Hello, I have checked it with fresh project and even with production live project in debug and release mode. No errors, Android/iOS is assembling correctly

**Comment by oleksandr-dziuban** [comment]
> Regarding [this](https://github.com/react-native-community/react-native-share/issues/549#issuecomment-513048235) comment, we should add instructions about [jetifier](https://github.com/mikehardy/jetifier) on README, as well.

Jetifier is always enabled if user migrated to react-native@0.60+. It is added as enable.jetifier=true property in gradle settings.

**Comment by mikehardy** [comment]
@oleksandr-dziuban I said:

```
and then people that are stuck on RN0.59 should be advised to use jetifier -r as a package.json postinstall.
```

Your comment does not address that.

There are two jetifiers. There is the Java-level gradle tool you reference. You are correct it is always on for RN0.60, and the react-native-community/cli authors even worked with me to add the second tool - the javascript-level node_modules jetifier - to the CLI so it is on by default as well.

But what do you do when you are stuck on RN0.59 and you want to use the latest react-native-share after your breaking change?

Consider that.

That is what `jetifier -r` is for. It reverses the mapping then the transition is easy (by making modern library versions compatible with android support libraries) for people that can't move quite yet.

That's what I meant in my comment

**Comment by oleksandr-dziuban** [comment]
> @oleksandr-dziuban I said:
> 
> ```
> and then people that are stuck on RN0.59 should be advised to use jetifier -r as a package.json postinstall.
> ```
> 
> Your comment does not address that.
> 
> There are two jetifiers. There is the Java-level gradle tool you reference. You are correct it is always on for RN0.60, and the react-native-community/cli authors even worked with me to add the second tool - the javascript-level node_modules jetifier - to the CLI so it is on by default as well.
> 
> But what do you do when you are stuck on RN0.59 and you want to use the latest react-native-share after your breaking change?
> 
> Consider that.
> 
> That is what `jetifier -r` is for. It reverses the mapping then the transition is easy (by making modern library versions compatible with android support libraries) for people that can't move quite yet.
> 
> That's what I meant in my comment

Ok, I see, thanks. 
So what are the next steps? You don't want to merge this PR?

**Comment by mikehardy** [comment]
I have been clear on my preference - the PR is the right direction, assuming it's good (I've not tested it), merging an AndroidX transition PR is the right idea.

The next step is just to update documentation for people still on react-native 0.59, tell them they will need `jetifier -r` in their postinstall after this. This will be a common recommendation across lots of libraries

**Comment by MateusAndrade** [comment]
> I have been clear on my preference - the PR is the right direction, assuming it's good (I've not tested it), merging an AndroidX transition PR is the right idea.
> 
> The next step is just to update documentation for people still on react-native 0.59, tell them they will need `jetifier -r` in their postinstall after this. This will be a common recommendation across lots of libraries

I can handle this update and the test. :smile: 

But how someone using rn 0.59 would need to updated their project? The will need to enable jetifier? This is not clear to me.

**Comment by mikehardy** [comment]
If you read this you'll understand, and you can link to it, https://github.com/mikehardy/jetifier/blob/master/README.md#to-reverse-jetify--convert-node_modules-dependencies-to-support-libraries

**Comment by MateusAndrade** [comment]
Just tested on a project using the following: 

```json
{
    "react-native": "0.59.2",
    "react-native-animatable": "^1.3.2",
    "react-native-camera": "^1.10.0",
    "react-native-confirmation-code-input": "^1.0.4",
    "react-native-firebase": "5.2.1",
    "react-native-gesture-handler": "^1.0.15",
    "react-native-keyboard-aware-scroll-view": "^0.8.0",
    "react-native-linear-gradient": "^2.5.3",
    "react-native-masked-text": "^1.10.1",
    "react-native-orientation-locker": "^1.1.2",
    "react-native-picker-select": "^6.0.0",
    "react-native-share": "git+https://github.com/oleksandr-dziuban/react-native-share",
    "react-native-splash-screen": "^3.2.0",
    "react-native-tab-view": "^1.3.1",
    "react-native-vector-icons": "^6.2.0",
    "react-native-view-shot": "^2.6.0",
    "react-navigation": "^3.0.9"
}
```

And it works for dev mode using ```react-native run-android``` and for release without any problem. i just needed to run a ```npx jetify -r``` after running a ```yarn``` and it build for both modes like a charm.  I also tested the share functions using both dev/release build and it works perfectly. :tada: :tada: :tada: 

Btw, thanks @mikehardy for saving our lifes with jetifier. :smile:

File: README.md [hunk 1]
```diff
@@ -13,6 +13,30 @@ This gives you the power to prioritize our work and support the project contribu
...
+#### Important:
+Linking is not needed anymore. ``react-native@0.60.0+`` supports dependencies auto linking.
+For iOS you also need additional step to install auto linked Pods (Cocoapods should be installed):
+``` 
+cd ios && pod install && cd ../
+```
+___
+
+### If you are using `react-native` <= 0.59.10 please use `react-native-share` <= 1.2.1:
+
```
Comment (MateusAndrade) [inline README.md]: Would be interesting warning about the possible problems using this PR. Adding something like: 

```suggestion
If you are having any problems with this library, or need to use >= 2.0.0 please refer to: [jetifier](https://github.com/mikehardy/jetifier#to-reverse-jetify--convert-node_modules-dependencies-to-support-libraries).

After installing jetifier, runs a ```npx jetify -r``` and test if this works by running a ```react-native run-android```.
```

**Comment by MateusAndrade** [comment]
Its looks ok for me now. Do you think we can merge this @mikehardy?

**Comment by MateusAndrade** [comment]
Thanks for helping with that @oleksandr-dziuban! 

@mikehardy  what do you think about pin a issue wih the possible problems related to this update?

**Comment by mikehardy** [comment]
@MateusAndrade maybe? :-) - in react-native-firebase we pinned one because the issue volume was crazy. I would leave it to you, if you notice you're spending a lot of time on AndroidX chatter, pin one. Can't really hurt. Nothing else super-hot in this project I think so it's not as if the pin area is too full?

**Pull request merged**